plugins/experimental/ts_lua/ts_lua_fetch.c

```diff
@@ -260,9 +260,12 @@ ts_lua_fetch_one_item(lua_State *L, const char *url, size_t url_len, ts_lua_fetc
     lua_gettable(L, -2);
 
     if (lua_isstring(L, -1)) {
-      addr = luaL_checklstring(L, -1, &addr_len);
-      rc   = sscanf(addr, "%15s:%d", ipstr, &port);
-      if (rc == 2) {
+      addr          = luaL_checklstring(L, -1, &addr_len);
+      const char *p = strstr(addr, ":");
+      if (p && p - addr < 31) {
+        port = atoi(p + 1);
```

Comment:
perhaps it is better to check addr_len as well in determining the port. i.e. use the whole string from p+1 to addr+addr_len